### DISCUSSION TITLE
Loading I/Q files to start pipeline?

### DISCUSSION THREAD

**Discussion by evm-sec** [ainfosec/FISSURE#37] [Hard Questions]
I'm trying to demodulate some captures from a signal-of-interest.  I have IQ capture files from a HackRF Portapack.  Is there a way to load these captures into the pipeline?  I'm unclear how to use FISSURE without a physical capture device doing the capturing (it looks like you need to select a hardware device at each stage...leaving it set as "Computer" for TSI or PD doesn't seem to give me a place to load I/Q files from disk).

**Comment by cpoore1** [comment]
There is not much analysis available as of today. The Protocol Discovery (PD) component is going to be expanded to accept IQ recordings containing isolated signals and classification results obtained by the Target Signal Identification (TSI) component (looking only at the signals and not demodulating anything). There will be a place to iterate through these signals of interest manually or automatically while also having the ability to forego the TSI component and load the IQ files on your own right into PD. The PD component will evaluate the files and select flow graphs to demodulate the bits if it is believed to be a known protocol and try to load flow graphs recursively to get closer to a bitstream if it is an unknown protocol. We will try to support offline analysis and live collection with SDRs for this.

Currently, one can integrate a demodulation flow graph into the library that already does everything and pipe the bits over ZMQ to a circular buffer for some very limited analysis. The IQ Data tab exists to do some basic analysis on the files, but we are also looking for more measurement techniques to aid with demodulation. There's a lot to expand upon and we are open to ideas and techniques as we build it out.
